lms/static/js/student_account/views/account_settings_factory.js

```diff
@@ -191,6 +123,117 @@
                 }
             ];
 
+            if (!isShibAuth) {
+                aboutSectionsData[0].fields.push(
```

Comment:
Do we actually have to remove all of these and then append most of them back unchanged?

Could we just leave upstream there and then selectively override specific indices?
It seems like if we can use the email feature flag, then we only need to modify the full name field and remove password, right?

Could we do that like this?

```js
if (isShibAuth) {
    var password_index = 3;
    var fullname_index = 1;
    var fields = aboutSectionsData[0].fields;
    fields[fullname_index] = {
        view: new AccountSettingsFieldViews.ReadonlyFieldView({
            model: userAccountModel,
            title: gettext('Full Name'),
            valueAttribute: 'name',
            helpMessage: gettext(
                'The name that appears on your Statements of Accomplishment. Other learners never see your full name.'  // eslint-disable-line max-len
            )
        })
    };
    fields.splice(password_index, 1);
}
```